racksdb/web/ui/src/components/InfrastructureTable.vue

```diff
@@ -10,364 +10,239 @@ import type {
-  showInfrastructureRacks.value = false
-  var layout = props.infrastructureDetails.layout
+type EquipmentType = 'nodes' | 'storage' | 'network' | 'misc'
+
+interface Node extends NodeEquipment {
+  equipmentType: EquipmentType
+}
+
+interface Network extends NetworkEquipment {
+  equipmentType: EquipmentType
+}
+
+interface Storage extends StorageEquipment {
+  equipmentType: EquipmentType
+}
+
+interface Misc extends MiscEquipment {
+  equipmentType: EquipmentType
+}
 
-  for (let index = 0; index < layout!.length; index++) {
-    rack.value = layout!.filter((rack) => rack.rack === rackName)
+function displayRackEquipment(rackName: string) {
+  const index = rackNames.value.indexOf(rackName)
+  if (index !== -1) {
+    rackDisplayStates.value[index] = !rackDisplayStates.value[index]
   }
 }
 
-function allInfrastructurDetails() {
-  showRack.value = false
-  showInfrastructureRacks.value = true
+function sortRacks() {
+  if (rackNames.value) {
+    rackNames.value.reverse()
+    rackDisplayStates.value.reverse()
+  }
+  alphabeticalOrder.value = !alphabeticalOrder.value
 }
 
-function popUp(name: string, equipment: equipmentType) {
+function getRacks() {
+  var layout = props.infrastructureDetails.layout
+
+  layout.forEach((rack) => {
+    rackNames.value.push(rack.rack)
+  })
+}
+
+// This function get in parameter the name of the rack, filter the layout with the rack name and return the result
+function getRackEquipments(rackName: string) {
+  let equipments: Array<Node | Network | Storage | Misc> = []
+  const parts = props.infrastructureDetails.layout.filter((part) => part.rack == rackName)
+
+  /*
+   * For every parts, insert all nodes, network equipments, etc... enriched
+   * with equipmentType in equipments array.
+   */
+  parts.forEach((part) => {
+    equipments.push(
+      ...part.nodes.map((node) => {
+        return { ...{ equipmentType: 'nodes' }, ...node } as Node
+      })
+    )
+    equipments.push(
+      ...part.network.map((network) => {
+        return { ...{ equipmentType: 'network' }, ...network } as Network
+      })
+    )
+    equipments.push(
+      ...part.storage.map((storage) => {
+        return { ...{ equipmentType: 'storage' }, ...storage } as Storage
+      })
+    )
+    equipments.push(
+      ...part.misc.map((misc) => {
+        return { ...{ equipmentType: 'misc' }, ...misc } as Misc
+      })
+    )
+  })
+  // Sort equipment by slot in descending order
+  return equipments.sort((a, b) => b.position.height - a.position.height)
+}
+
+function popUp(name?: string, equipmentType?: EquipmentType) {
   if (showPopUp.value) {
     showPopUp.value = !showPopUp.value
   } else {
     var layout = props.infrastructureDetails.layout
 
-    for (let x = 0; x < layout.length; x++) {
-      var result = layout[x][equipment].find((item) => item.name == name)
-      if (result) {
-        popUpContent.value = result
-        break
+    if (equipmentType) {
+      for (let x = 0; x < layout.length; x++) {
```

Comment:
You can avoid this equipment search in the infrastructure by directly passing the modal content in parameter of the function, as you already have all the context when you call this function in the template.